lib/spotify_sdk_web.dart

```diff
@@ -463,18 +497,38 @@ class SpotifySdkPlugin {
 
     // exchange auth code for access and refresh tokens
     dynamic authResponse;
+
+    // build request to exchange auth code with PKCE for access and refresh tokens
+    var req = RequestOptions(
+      path: 'https://accounts.spotify.com/api/token',
+      method: 'POST',
+      data: {
+        'client_id': clientId,
+        'grant_type': 'authorization_code',
+        'code': parsedMessage.queryParameters['code'],
+        'redirect_uri': redirectUrl,
+        'code_verifier': codeVerifier
+      },
+      contentType: Headers.formUrlEncodedContentType,
+    );
+
+    // or build request to exchange code with token swap
+    // https://developer.spotify.com/documentation/ios/guides/token-swap-and-refresh/
+    if (_tokenSwapURL != null) {
+      req = RequestOptions(
+        path: _tokenSwapURL!,
+        method: 'POST',
+        data: {
+          'code': parsedMessage.queryParameters['code'],
+          'redirect_uri': redirectUrl,
+        },
+        contentType: Headers.formUrlEncodedContentType,
+      );
+    }
```

Comment:
RequestOptions req;
```suggestion
    if (_tokenSwapURL == null) {
      // build request to exchange auth code with PKCE for access and refresh tokens
    req = RequestOptions(
      path: 'https://accounts.spotify.com/api/token',
      method: 'POST',
      data: {
        'client_id': clientId,
        'grant_type': 'authorization_code',
        'code': parsedMessage.queryParameters['code'],
        'redirect_uri': redirectUrl,
        'code_verifier': codeVerifier
      },
      contentType: Headers.formUrlEncodedContentType,
    );
    }
    else {
        // or build request to exchange code with token swap
        // https://developer.spotify.com/documentation/ios/guides/token-swap-and-refresh/
      req = RequestOptions(
        path: _tokenSwapURL!,
        method: 'POST',
        data: {
          'code': parsedMessage.queryParameters['code'],
          'redirect_uri': redirectUrl,
        },
        contentType: Headers.formUrlEncodedContentType,
      );
    }
```

Let's use if-else here to make the 2 paths more explicit. I messed up the indentation and needs a couple of fixes for null safety but I hope you get the idea.

Reply:
fixed